src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int INCOME_INDEX = 3;
+    private static final String SPLITERATOR = " ";
+    private static final String DATE_FORMAT = "dd.MM.yyyy";
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern(
+            DATE_FORMAT);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salarySum = new int[names.length];
+        StringBuilder result = new StringBuilder(
+                "Report for period " + dateFrom + " - " + dateTo);
+
+        LocalDate parseDateFrom = LocalDate.parse(dateFrom, DATE_FORMATTER);
+        LocalDate parseDateTo = LocalDate.parse(dateTo, DATE_FORMATTER);
+
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                String[] dataInfo = data[j].split(SPLITERATOR);
+                LocalDate dataDate = LocalDate.parse(dataInfo[DATE_INDEX], DATE_FORMATTER);
+                if ((dataDate.isAfter(parseDateFrom)
```

Comment:
It is very difficult to read this boolean expression, please extract it to method. Also there are no sense in 2 if statement in a row you can do everything in one if

Reply:
thx! fixed